xbmc/platform/PlatformFactory.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ *      Copyright (C) 2016 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Kodi; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "PlatformFactory.h"
+#include "interfaces/IPlatform.h"
+#if defined(TARGET_DARWIN)
+#include "platform/darwin/PlatformDarwin.h"
+#endif
+
+IPlatform *CPlatformFactory::GetPlatform()
+{
+#if defined(TARGET_DARWIN)
```

Comment:
yes but the automagically setting of the needed define wouldn't workt then like done here for example:

https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/Process/CMakeLists.txt#L1

because it looks for the file in CORE_SYSTEM_NAME which would be darwin in that case